src/coreComponents/dataRepository/Group.cpp

```diff
@@ -28,6 +28,9 @@ namespace geosx
 namespace dataRepository
 {
 
+/// @brief An empty orphan group node for null-value checks by address
+static Group NullGroup("null",nullptr);
```

Comment:
I'm not super fond of this mechanism but its the least invasive way to support a `hasGroupByPath` implementation.